lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -1292,9 +1333,9 @@ def pretty_instance(i):
 def determine_iam_role(name_or_arn):
     if re.match(r'^arn:aws:iam::\d+:instance-profile/[\w+=/,.@-]+$', name_or_arn):
         return name_or_arn
-    iam = module.client('iam')
+    iam = module.client('iam', retry_decorator=AWSRetry.jittered_backoff())
     try:
-        role = iam.get_instance_profile(InstanceProfileName=name_or_arn)
+        role = iam.get_instance_profile(InstanceProfileName=name_or_arn, aws_retry=True)
```

Comment:
We should update the [GUIDELINES](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#api-throttling-and-pagination) if we're happy enough to use it now (it's so much nicer than creating `_backoff` methods with `@AWSRetry` decorators)

Reply:
I'm quite happy with it (so nice to use!) but s-hertel brought up an interesting point about it for library/common functions. In module_utils files, we probably can't expect the caller to always have a retry decorator turned on for whatever client they pass in, so I think for utils we still have to use the decorators directly for now. What do you think?